### PR TITLE
Button: Icon placement setting

### LINKED ISSUE #440: Button widget: Icon placement setting
[Suggested here](https://siteorigin.com/thread/site-origin-widget-button/)

A really nice addition to the SiteOrigin Button widget would be the ability to easily dictate the placement of the icon (relative to the text) via a drop down - top, left, right and bottom.

### PR DESCRIPTION
**Pull request by AlexGStapleton**
Resolves #440

### CODE REVIEW

**Comment by AlexGStapleton** [comment]
Sorry about that! 😞 
I'll commit the rest of the files in a second.

**Comment by AlexGStapleton** [comment]
There you go - sorry about that!

File: widgets/button/button.php [hunk 1]
```diff
@@ -231,14 +243,15 @@ function get_template_variables( $instance, $args ) {
 
 		$attributes = $instance['attributes'];
 
+		// 'icon_placement' => $instance['button_icon']['icon_placement'],
+
 		$classes = ! empty( $attributes['classes'] ) ? $attributes['classes'] : '';
+		$classes .= ' ow-icon-placement-'. $instance['button_icon']['icon_placement'];
```
Comment (braamgenis) [inline widgets/button/button.php]: Will need an `empty` check here.

File: widgets/button/css/style.less [hunk 2]
```diff
@@ -11,20 +11,38 @@
 		text-decoration: none;
 		line-height: 1em;
 
+		&.ow-icon-placement-top {
```
Comment (braamgenis) [inline widgets/button/css/style.less]: The top setting doesn't seem to work for me. The other all work fine though.

**Comment by AlexGStapleton** [comment]
Sorry about that. 😖

**Pull request merged**